samples/dotnet/github-skills/GitHubSkill.cs

```diff
@@ -37,6 +36,11 @@ public static class Parameters
         /// </summary>
         public const string SearchPattern = "searchPattern";
 
+        /// <summary>
+        /// Personal access token for private repositories.
+        /// </summary>
+        public const string PatToken = "patToken";
```

Comment:
I don't think this is actionable with the current architecture, since the front-end app doesn't have direct access to the Kernel, but flagging anyway...

In general: I don't think we should be putting passwords into ContextVariables. Instead, the Skill should be constructed with that password, or set as an env/config/app secret.

ContextVariables will get passed in plain text to the AI in various prompts, and be visible to all other functions, etc.

In this application, I also don't think the GitHub logic needs to be in a Skill at all, since the AI is not calling it. 
SK team: should we remove the GitHubSkill and just put this in a helper function? @hathind-ms @SergeyMenshykh @dluc @adrianwyatt

Reply:
I agree, although I was not sure what is a sample (to guide an eventual implementation) and what is for demo, there are other security issues out there, the most visible is that the OpenAPI key is passed in plain text from the front-end to the backend (although I saw signs of this probably being migrated to env vars).

FWIW The front-end does not have direct access to the Kernel, but it does through function apps (the kernel http service). This code actually works, I have tested this with private repo.

Reply:
Hey @mahomedalid sorry for the delay. I agree, and I did not intend to block this PR for the above issue. I was just pointing it out to peers, as it's a thing we should look out for: passing private keys as ContextVariables.

Though not strictly the case, SKFunctions are intended to be available for the AI to call via Planners, and ContextVariables are how the data moves between them. At times, the ContextVariables can be sent in plain text to the AI, and they are available to all other functions in the Plan.

The issue you've (accidentally) uncovered above is just good for us to know about so we can help communicate best practices.

This is a great addition to the demo, and we'd like to merge it in. Once you do, we'll have somebody look at reworking how that skill gets its keys. If you don't mind, could you please sync with main, resolve conflicts, and run the `build.cmd` command under the repo root? The integration test may fail if you don't have keys set up, but the unit tests should all pass, and dotnet format will fix the errors showing up in the checks below.

Thanks again!

Reply:
@shawncal yeah, don't worry, I understand. Solved conflicts, run build.sh, I saw some integration tests failing as you mentioned, otherwise looks good.

```

Failed!  - Failed:    24, Passed:    27, Skipped:     6, Total:    57, Duration: 1 s - IntegrationTests.dll (net6.0)

Passed!  - Failed:     0, Passed:   632, Skipped:     0, Total:   632, Duration: 1 s - SemanticKernel.UnitTests.dll (net6.0)

Passed!  - Failed:     0, Passed:    90, Skipped:     0, Total:    90, Duration: 4 s - SemanticKernel.Connectors.UnitTests.dll (net6.0)

Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 380 ms - SemanticKernel.Skills.UnitTests.dll (net6.0)
```